### DISCUSSION TITLE
Template notebooks

### DISCUSSION THREAD

**Discussion by BioTurboNick** [fonsp/Pluto.jl#2256] [Pluto development]
Similar thinking to #550 but more general.

I'd like to provide users with multiple template notebooks they can select from. The idea would be that they would open a copy of a notebook and then save it in another location.

For a use case, we have a basic analysis workflow to be reused to generate a report from each experiment. Currently, people have to copy the template in a terminal and then open it in Pluto. Once a notebook is open, the only option is to move it to a new location.

**Comment by disberd** [comment]
I am not sure I fully understand your use case (It is not entirely clear what the users have to paste in a terminal in your current scenario), but what about hosting your template notebooks online (like on github) and then have the users just put the template notebook URL in the `open a notebook` field on the Pluto welcome page?
That would create a temporary notebook which is a copy of the one in the URL that users can then modify and save elsewhere.

Alternatively, if you have to keep things local and without internet, why not just copying the template directly from the REPL to the intended new location and then just open that as a normal notebook to be modified?

**Comment by BioTurboNick** [comment]
> Alternatively, if you have to keep things local and without internet, why not just copying the template directly from the REPL to the intended new location and then just open that as a normal notebook to be modified?

That's what they're doing now. But also they sometimes copy a previous notebook rather than work from the template. And I'd like it to be something that doesn't require working with the shell.

> I am not sure I fully understand your use case (It is not entirely clear what the users have to paste in a terminal in your current scenario), but what about hosting your template notebooks online (like on github) and then have the users just put the template notebook URL in the `open a notebook` field on the Pluto welcome page? That would create a temporary notebook which is a copy of the one in the URL that users can then modify and save elsewhere.

I suppose that would work, though they'd still have to look up the URL each time. I was in part inspired by the nice new Featured Notebooks UI. I'm imagining there could be a "Templates" section that could be customized, and users can just tap the image to get a notebook ready for use for a specific purpose.

Maybe I can explain the use case a bit better - there's a standard data path, the user points a variable at the subfolder containing the data they want to analyze, then all our standard plots and tables are generated.

**Comment by disberd** [comment]
I see your point now. Indeed it would be nice to have maybe a default folder where present notebooks are automatically shown in the welcome page like template ones like it was suggested in #550.
